base/experimental.jl

```diff
@@ -321,7 +321,7 @@ function show_error_hints(io, ex, args...)
             @invokelatest handler(io, ex, args...)
         catch err
             tn = typeof(handler).name
-            @error "Hint-handler $handler for $(typeof(ex)) in $(tn.module) caused an error"
+            @error "Hint-handler $handler for $(typeof(ex)) in $(tn.module) caused an error" err
```

Comment:
IIRC, if you name this _exception it will get better printing, and if you pass _exception=current_exceptions() it will do even better (though also will include stacktraces, so YMMV)

Reply:
I intentionally didn't include the stacktrace but I'll add it. Though `_exceptions` doesn't seem to be special (but I do see a couple of uses of it in Base/REPL)

```
julia> try
       error("some error")
       catch e
       @error "log error" _exception=current_exceptions()
       end
┌ Error: log error
│   _exception =
│    1-element ExceptionStack:
│    some error
│    Stacktrace:
│      [1] error(s::String)
│        @ Base ./error.jl:44
│      [2] top-level scope
│        @ REPL[7]:2
│      [3] __repl_entry_eval_expanded_with_loc(mod::Module, ast::Any, toplevel_file::Ref{Ptr{UInt8}}, toplevel_line::Ref{Int32})
│        @ REPL ~/Documents/GitHub/julia/usr/share/julia/stdlib/v1.12/REPL/src/REPL.jl:331
...
```
```
julia> try
       error("some error")
       catch e
       @error "log error" exception=current_exceptions()
       end
┌ Error: log error
│   exception =
│    1-element ExceptionStack:
│    some error
│    Stacktrace:
│      [1] error(s::String)
│        @ Base ./error.jl:44
│      [2] top-level scope
│        @ REPL[8]:2
│      [3] __repl_entry_eval_expanded_with_loc(mod::Module, ast::Any, toplevel_file::Ref{Ptr{UInt8}}, toplevel_line::Ref{Int32})
│        @ REPL ~/Documents/GitHub/julia/usr/share/julia/stdlib/v1.12/REPL/src/REPL.jl:331
...
```